logging/logger.go

```diff
@@ -106,6 +107,20 @@ func FromContext(ctx context.Context) *zap.SugaredLogger {
 	return Default()
 }
 
+// TestLogger returns a logger configured for tests. It will only output log
+// information if specific test fails or is run in verbose mode. See [zaptest]
+// for more information.
+//
+//	func TestMyThing(t *testing.T) {
+//		logger := logging.TestLogger(t)
+//		thing := &MyThing{logger: logger}
+//	}
+//
+// [zaptest]: https://pkg.go.dev/go.uber.org/zap/zaptest
+func TestLogger(tb zaptest.TestingT) *zap.SugaredLogger {
```

Comment:
Optional: consider accepting a parameter "opts ...zaptest.LoggerOption" to this function, and passing them through to zaptest.NewLogger. That would let the caller override the level. Dunno if we actually care about that though.